### ISSUE TITLE
Problem with running vignette

### ISSUE THREAD

**Issue by ghost** [mfasiolo/mgcViz#30]
Hi

```r
library(mgcViz)
n  <- 1e3
x1 <- rnorm(n)
x2 <- rnorm(n)
dat <- data.frame("x1" = x1, "x2" = x2,
                  "y" = sin(x1) + 0.5 * x2^2 + pmax(x2, 0.2) * rnorm(n))
b <- gam(y ~ s(x1, x2), data = dat, method = "REML")
b <- getViz(b)
plot(sm(b, 1)) + l_fitRaster() + l_fitContour() + l_points()
```

The plot returns an error : 

```
Error in UseMethod("rescale") : 
  pas de méthode pour 'rescale' applicable pour un objet de classe "AsIs"
```

Here is my session Info (I suspect a package version problem):

```
> sessionInfo()
R version 3.4.1 (2017-06-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] fr_FR.UTF-8/fr_FR.UTF-8/fr_FR.UTF-8/C/fr_FR.UTF-8/fr_FR.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] mgcViz_0.0.1  rgl_0.98.1    ggplot2_2.2.1 mgcv_1.8-22   nlme_3.1-131 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.13       compiler_3.4.1     RColorBrewer_1.1-2 plyr_1.8.4         bindr_0.1         
 [6] viridis_0.4.0      tools_3.4.1        mvnfast_0.2.2      digest_0.6.12      viridisLite_0.2.0 
[11] jsonlite_1.5       tibble_1.3.4       gtable_0.2.0       lattice_0.20-35    pkgconfig_2.0.1   
[16] rlang_0.1.2        Matrix_1.2-10      rstudioapi_0.7     shiny_1.0.5        GGally_1.3.2      
[21] yaml_2.1.14        bindrcpp_0.2       gridExtra_2.3      httr_1.3.1         dplyr_0.7.4       
[26] knitr_1.16         htmlwidgets_0.9    grid_3.4.1         glue_1.1.1         reshape_0.8.7     
[31] data.table_1.10.4  R6_2.2.2           plotly_4.7.1       tidyr_0.7.1        purrr_0.2.2.2     
[36] magrittr_1.5       matrixStats_0.52.2 scales_0.5.0       htmltools_0.3.6    assertthat_0.2.0  
[41] mime_0.5           xtable_1.8-2       colorspace_1.3-2   httpuv_1.3.5       KernSmooth_2.23-15
[46] miniUI_0.1.1       lazyeval_0.2.0     munsell_0.4.3     
``

**Comment by mfasiolo** [comment]
This seems to be due to the fact that `plot(sm(b, 1))` uses `trans = I` by default, where `I()` returns an object of class "AsIs". But `geom_raster` expects an numeric vector or a list. Quick fix is adding

`if( class(a$data$tz) == "AsIs" ) { class(a$data$tz) <- NULL }`

which is implemented here a334b107237eeb87760402ad9bc0e5cedf45ed0a. Seems to work, but not sure whether this is the definitive solution.

**Comment by mfasiolo** [comment]
Indeed the last solution was not good enough. I substituted I() with function(.x){ .x } every where, as I don't see why we really need I(). Done here 21745bbf8b22ab9f76d3f806831288f6fa192873
